examples/distinct_count_estimator/host_bulk_example.cu

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <cuco/distinct_count_estimator.cuh>
+
+#include <thrust/device_vector.h>
+#include <thrust/sequence.h>
+
+#include <cstddef>
+#include <iostream>
+
+int main()
+{
+  using T                         = int;
+  std::size_t constexpr num_items = 1ull << 30;  // 4GB
+
+  thrust::device_vector<T> items(num_items);
+  // create a vector of distinct items
+  thrust::sequence(items.begin(), items.end(), 0);
+
+  cudaEvent_t start, stop;
+  cudaEventCreate(&start);
+  cudaEventCreate(&stop);
+
+  cuco::distinct_count_estimator<T> estimator;
+  cudaEventRecord(start);
+  // add all items to the estimator
+  estimator.add(items.begin(), items.end());
+  // after the estimator has seen all items, we can calculate the cardinality
+  std::size_t const estimated_cardinality = estimator.estimate();
+  cudaEventRecord(stop);
+  cudaEventSynchronize(stop);
+
+  float milliseconds = 0;
+  cudaEventElapsedTime(&milliseconds, start, stop);
```

Comment:
question: Do we intend to leave the timing code in the example? I think it would be cleaner to keep separate examples and benchmarks.

Reply:
Agreed. This is just a temporary thing until I have implemented proper benchmarks. I'll remove it before I mark this PR as ready.